[SEMANTIC CONVENTIONS] Upgrade to 1.21.0

marcalff opened the pull request:
Fixes #2138

## Changes

Upgrade:

* Semantic conventions to 1.21.0
* Build tools to 0.19.0

Adjust code per the following semantic convention changes:

* https://github.com/open-telemetry/opentelemetry-specification/pull/3402
* https://github.com/open-telemetry/opentelemetry-specification/pull/3355

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Unit tests have been added
* [X] Changes in public API reviewed

lalitb reviewed (approved):
LGTM Thanks for the upgrade. Not related to this PR, but the trace and resource header files are quite big now. Also, someone interested in only DB-specific instrumentation will have HTTP-related static variables getting created and never used.
You know this part better, but is it feasible to generate domain-specific header files, say `trace/semantic_convention_db.h`, `trace/semantic_convention_http.h` etc? We can still have the existing `trace/semantic_convention.h` which includes all these domain-specific header files. If feasible, we can have an issue created for someone to pick.

marcalff commented:
> LGTM Thanks for the upgrade. Not related to this PR, but the trace and resource header files are quite big now. Also, someone interested in only DB-specific instrumentation will have HTTP-related static variables getting created and never used. You know this part better, but is it feasible to generate domain-specific header files, say `trace/semantic_convention_db.h`, `trace/semantic_convention_http.h` etc? We can still have the existing `trace/semantic_convention.h` which includes all these domain-specific header files. If feasible, we can have an issue created for someone to pick.

This can be investigated.

The risk I see is that the build scripts and the header file layout produced will be more coupled to the internal code organization in the semantic-convention/model directory:

* If a new yaml file is added, we may not notice it, forgetting to generate semconv for it
* If a yaml file is split, moved, merged, renamed, etc, the header files generated for it will be affected, breaking C++ users who include this header file alone.

I would rather wait for a couple of semantic-conventions releases first, to make sure the code layout itself is stable, before we can consider this.
